source/simulator/initial_conditions.cc

```diff
@@ -375,84 +375,30 @@ namespace aspect
+                                                                    [&](const typename DoFHandler<dim>::active_cell_iterator & /*cell*/,
+                                                                        const std::vector<Point<dim> > &q_points,
+                                                                        std::vector<double> &values) -> void
         {
-          return adiabatic_conditions->pressure(p);
-        });
-
-
-        for (const auto &cell : dof_handler.active_cell_iterators())
-          if (cell->is_locally_owned())
-            {
-              cell->get_dof_indices (local_dof_indices);
-              fe_values.reinit(cell);
-
-              adiabatic_pressure.value_list (fe_values.get_quadrature_points(),
-                                             rhs_values);
-
-              cell_vector = 0;
-              local_mass_matrix = 0;
-              for (unsigned int point=0; point<n_q_points; ++point)
-                for (unsigned int i=0; i<dofs_per_cell; ++i)
-                  {
-                    if (finite_element.system_to_component_index(i).first == dim)
-                      cell_vector(i)
-                      +=
-                        rhs_values[point] *
-                        fe_values[introspection.extractors.pressure].value(i,point) *
-                        fe_values.JxW(point);
-
-                    // populate the local matrix; create the pressure mass matrix
-                    // in the pressure pressure block and the identity matrix
-                    // for all other variables so that the whole thing remains
-                    // invertible
-                    for (unsigned int j=0; j<dofs_per_cell; ++j)
-                      if ((finite_element.system_to_component_index(i).first == introspection.component_indices.pressure)
-                          &&
-                          (finite_element.system_to_component_index(j).first == introspection.component_indices.pressure))
-                        local_mass_matrix(j,i) += (fe_values[introspection.extractors.pressure].value(i,point) *
-                                                   fe_values[introspection.extractors.pressure].value(j,point) *
-                                                   fe_values.JxW(point));
-                      else if (i == j)
-                        local_mass_matrix(i,j) = 1;
-                  }
-
-              // now invert the local mass matrix and multiply it with the rhs
-              local_mass_matrix.gauss_jordan();
-              local_mass_matrix.vmult (local_projection, cell_vector);
-
-              // then set the global solution vector to the values just computed
-              cell->set_dof_values (local_projection, system_tmp);
-            }
+          for (unsigned int i=0; i<values.size(); ++i)
+            values[i] = adiabatic_conditions->pressure(q_points[i]);
+          return;
+        },
+        system_tmp);
```

Comment:
why is this not ``system_tmp.block(...)``?

Reply:
The input Vector  needs to match the DoFHandler for `set_dof_value()` to correctly fill the vector.